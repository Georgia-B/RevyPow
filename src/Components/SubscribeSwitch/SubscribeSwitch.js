import React from 'react';
import Switch from '@material-ui/core/Switch';
import FormControlLabel from '@material-ui/core/FormControlLabel';

const SubscribeSwitch = (props) => {
    return (
        <div className="subscribe__container">
            <FormControlLabel
                control={
                    <Switch
                        checked={props.isSubscribed}
                        onChange={props.onClick}
                        value={props.isSubscribed}
                        color="secondary"
                        inputProps={{ 'aria-label': 'primary checkbox' }}
                    />
                }
                label="Notifications"
            />
            {props.isSubscribed && <div className="subscribe__message">
                You are subscribed to the morning snow report.
            </div>}
        </div>
    );
}

export default SubscribeSwitch;