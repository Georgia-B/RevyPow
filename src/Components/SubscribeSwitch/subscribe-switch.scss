.subscribe__container {
    display: flex;
    align-items: center;
    padding-left: 30px;
    flex: 2;
}

.subscribe__message {
    font-size: 13px;
}

@media (max-width: $breakpoint-medium) {
    .subscribe__container {
        padding-left: 15px;
    }
}

@media (max-width: $breakpoint-small) {
    .subscribe__container > .MuiFormControlLabel-root {
        margin-right: 5px;
    }
}