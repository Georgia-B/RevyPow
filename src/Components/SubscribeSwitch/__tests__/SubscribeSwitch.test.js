import React from 'react'
import { cleanup, render } from '@testing-library/react'
import SubscribeSwitch from '../SubscribeSwitch';

describe('Components | <SubscribeSwitch /> when subscribed', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<SubscribeSwitch isSubscribed={true} onClick={() => true} />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})

describe('Components | <SubscribeSwitch /> when not subscribed', () => {
    afterEach(cleanup);
    it('should render correctly', () => {
        const component = render(<SubscribeSwitch isSubscribed={false} onClick={() => true} />);
        expect(component.asFragment()).toMatchSnapshot();
    })
})