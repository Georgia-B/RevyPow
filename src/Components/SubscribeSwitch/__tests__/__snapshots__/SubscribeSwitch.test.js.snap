// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components | <SubscribeSwitch /> when not subscribed should render correctly 1`] = `
<DocumentFragment>
  <div
    class="subscribe__container"
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-93 MuiSwitch-switchBase MuiSwitch-colorSecondary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-label="primary checkbox"
              class="PrivateSwitchBase-input-96 MuiSwitch-input"
              type="checkbox"
              value="false"
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Notifications
      </span>
    </label>
  </div>
</DocumentFragment>
`;

exports[`Components | <SubscribeSwitch /> when subscribed should render correctly 1`] = `
<DocumentFragment>
  <div
    class="subscribe__container"
  >
    <label
      class="MuiFormControlLabel-root"
    >
      <span
        class="MuiSwitch-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-19 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-20 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              aria-label="primary checkbox"
              checked=""
              class="PrivateSwitchBase-input-22 MuiSwitch-input"
              type="checkbox"
              value="true"
            />
            <span
              class="MuiSwitch-thumb"
            />
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiSwitch-track"
        />
      </span>
      <span
        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
      >
        Notifications
      </span>
    </label>
    <div
      class="subscribe__message"
    >
      You are subscribed to the morning snow report.
    </div>
  </div>
</DocumentFragment>
`;
